Misc/NEWS.d/next/Core and Builtins/2020-12-02-20-23-31.bpo-42536.Kx3ZOu.rst

```diff
@@ -0,0 +1,2 @@
+Fix an issue where :func:`zip` may create untracked reference cyles. Patch
```

Comment:
Hum, it's unclear to me that the fix involved an internal tuple which is not tracked by the garbage collector. I suggest something like:

":func:`zip` now ensures that its internal tuple is always tracked by the garbage collector to be able to break reference cycle. Previously, the internal tuple was sometimes untracked by a garbage collection."

The reality is even more complex, it's an old version of the tuple which is untracked, a version which cannot be part of a reference cycle: a tuple only containing None values. But after the tuple is untracked, the tuple is modified in place (yeah, modifying an immutable object ;-)) and the new tuple can be involved in a reference cycle...

But I don't think that we should elaborate too much in the NEWS entry. People can look into the bpo for details ;-)